### PR TITLE
HPCC-12686 Use 3 enumerations in ws_dfu/ws_fs/ws_wu ecm files

### PR DESCRIPTION
**Pull request by wangkx**
1. in ws_dfu.ecm, change DFUArrayActionRequest.type to ESPenum;
2. in ws_fs.ecm, change DFUWorkunitsActionRequest.type to ESPenum;
3. in ws_workunits.ecm, change WUActionRequest.ActionType to ESPenum;
4. refactor the CFileSprayEx::onDFUWorkunitsAction().

Signed-off-by: wangkx kevin.wang@lexisnexis.com

### CODE REVIEW

**Comment by wangkx** [comment]
@afishbeck please review.

File: esp/services/ws_dfu/ws_dfuService.cpp [hunk 1]
```diff
@@ -1401,47 +1384,45 @@ bool CWsDfuEx::onDFUArrayAction(IEspContext &context, IEspDFUArrayActionRequest
...
-        if (strcmp(Action_AddtoSuperfile ,req.getType()) == 0)
+        if (errorStr.length())
         {
-            returnStr.replaceString("#", "%23");
-            if (version < 1.18)
-                resp.setRedirectUrl(StringBuffer("/WsDFU/AddtoSuperfile?Subfiles=").append(returnStr.str()));
-            else
-            {
-                resp.setRedirectTo(returnStr.str());
-            }
```
Comment (afishbeck) [inline esp/services/ws_dfu/ws_dfuService.cpp]: Does this mean we are no longer supporting Legacy ecl watch behavior?

Comment (afishbeck) [inline esp/services/ws_dfu/ws_dfuService.cpp]: Would kind of rather see these changes in a separate pull request... unless it has something to do with the switch to enumeration?

**Comment by afishbeck** [comment]
@wangkx looks good.

**Comment by wangkx** [comment]
@richardkchapman please merge this PR.

**Pull request merged**